<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Code - RecA Loop 2 evolutionary conservation</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Code";
    var mkdocs_page_input_path = "CODE.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> RecA Loop 2 evolutionary conservation</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../Loop2/">Loop2</a>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="./">Code</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#code-to-query-the-ncbi-database">Code to query the NCBI database</a></li>
    

    <li class="toctree-l2"><a href="#reciprocal-blast-of-the-reca-and-ku">Reciprocal BLAST of the RecA and Ku</a></li>
    

    <li class="toctree-l2"><a href="#cluster-and-calculate-pairwise-distances">Cluster and calculate pairwise distances</a></li>
    

    </ul>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">RecA Loop 2 evolutionary conservation</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Code</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h2 id="code-to-query-the-ncbi-database">Code to query the NCBI database</h2>
<pre><code class="matlab">% scReciprocalBLAST ---   script to start working from the reciprocal blast
% data of recA (MSMEG_2723) and ku (MSMEG_5580).

clear, close all; clc;
%     clears all workspaces

basedir         = '/Users/Dave/Desktop/recA_ku_orthologSearch/';
%     defines the base directory this script uses

[MSMEG_5580, locus5580, raw5580] = ...
    xlsread([basedir 'MSMEG_5580_orthologs.xls']);
[MSMEG_2723, locus2723, raw2723] = ...
    xlsread([basedir 'MSMEG_2723_orthologs.xls']);
clear locus2723 locus5580 raw2723 raw5580;
%     import the data. the two options denoted with locus and raw are
%     superfluous and therefore deleted

MSMEG_2723 = MSMEG_2723(2:end, [9 11]);
MSMEG_5580 = MSMEG_5580(2:end, [9 11]);
%     column 9 contains the taxID for the organisms; column 11 contains the
%     information about the NCBI gene ID. The first row is
%     header-information and is superfluous.

missingTaxID2723 = find(isnan(MSMEG_2723(:, 1)));
missingTaxID5580 = find(isnan(MSMEG_5580(:, 1)));
%     there are a few genes there there is no taxonomy ID
%     specified in the file. 

MSMEG_2723(missingTaxID2723, :) = [];
MSMEG_5580(missingTaxID5580, :) = [];
%     removing all the lines that have no mention of a taxonomy ID
    clear missingTaxID2723 missingTaxID5580

[C, ia, ib]   = intersect(MSMEG_2723, MSMEG_5580);
%     intersect the data sets to allow identification of only those strains
%     that have both ku and recA. This is merely to exclude any strains
%     that, through this way of searching their genomic data, are supposed
%     to only have ku.

ku          = MSMEG_5580(ib, :);
recAwithKu  = MSMEG_2723(ia, :);
recAnoKu    = MSMEG_2723;
    recAnoKu(ia, :) = [];
% recA        = MSMEG_2723;
%     parse out the data. ku maintains only those records that intersect
%     with recA, so those organisms have recA. Then there are three sets of
%     recA data: the full set (recA), the intersect with ku (recAwithKu)
%     and any recA that does not intersect with ku (recAnoKu).
    clear C ia ib MSMEG_2723 MSMEG_5580;

%% QUERY NC
baseURL         = 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=protein&amp;id=';
restURL         = '&amp;rettype=fasta&amp;retmode=text';
%     part FOUR of the URL to query, returning the accession codes
%         NOTE: the URL to submit for the query will consist of the base
%         URL, followed by the accID part, then the GI numbers with commas
%         in between them, and finished by rettype.
% samples2search = 1:length(recAnoKu);
GInumbers   = sprintf('%d,', recAnoKu(1:400, 2));
GInumbers   = char(GInumbers(1:end-1));
searchValue = strcat(baseURL, GInumbers, restURL);
% 
k = websave([basedir 'recAnoKu_protein.fasta'], searchValue); clear k
</code></pre>

<h2 id="reciprocal-blast-of-the-reca-and-ku">Reciprocal BLAST of the RecA and Ku</h2>
<pre><code class="matlab">% scReciprocalBLAST ---   script to start working from the reciprocal blast
% data of recA (MSMEG_2723) and ku (MSMEG_5580).

clear, close all; clc;
%     clears all workspaces

basedir         = '/Users/Dave/Desktop/recA_ku_orthologSearch/';
%     defines the base directory this script uses

[MSMEG_5580, locus5580, raw5580] = ...
    xlsread([basedir 'MSMEG_5580_orthologs.xls']);
[MSMEG_2723, locus2723, raw2723] = ...
    xlsread([basedir 'MSMEG_2723_orthologs.xls']);
clear locus2723 locus5580 raw2723 raw5580;
%     import the data. the two options denoted with locus and raw are
%     superfluous and therefore deleted

MSMEG_2723 = MSMEG_2723(2:end, [9 11]);
MSMEG_5580 = MSMEG_5580(2:end, [9 11]);
%     column 9 contains the taxID for the organisms; column 11 contains the
%     information about the NCBI gene ID. The first row is
%     header-information and is superfluous.

missingTaxID2723 = find(isnan(MSMEG_2723(:, 1)));
missingTaxID5580 = find(isnan(MSMEG_5580(:, 1)));
%     there are a few genes there there is no taxonomy ID
%     specified in the file. 

MSMEG_2723(missingTaxID2723, :) = [];
MSMEG_5580(missingTaxID5580, :) = [];
%     removing all the lines that have no mention of a taxonomy ID
    clear missingTaxID2723 missingTaxID5580

[C, ia, ib]   = intersect(MSMEG_2723, MSMEG_5580);
%     intersect the data sets to allow identification of only those strains
%     that have both ku and recA. This is merely to exclude any strains
%     that, through this way of searching their genomic data, are supposed
%     to only have ku.

ku          = MSMEG_5580(ib, :);
recAwithKu  = MSMEG_2723(ia, :);
recAnoKu    = MSMEG_2723;
    recAnoKu(ia, :) = [];
recA        = MSMEG_2723;
%     parse out the data. ku maintains only those records that intersect
%     with recA, so those organisms have recA. Then there are three sets of
%     recA data: the full set (recA), the intersect with ku (recAwithKu)
%     and any recA that does not intersect with ku (recAnoKu).
    clear C ia ib MSMEG_2723 MSMEG_5580;

%% QUERY KEGG FOR GENE IDS --- TIME CONSUMING!!
base            = 'http://rest.kegg.jp/';
operation       = 'conv/';
database        = 'genes/';
dbentry         = 'ncbi-gi:';

%%
kegg_ku = cell(length(ku), 1);
for i = 1:length(ku)
    dbentry1 = strcat(dbentry, num2str(ku(i, 2)));
    kegg_ku{i} = regexpi(urlread(strcat(base, operation, database,...
        dbentry1)), '(?&lt;=(??@dbentry1)\s+)\w+\W+\w*','match');
end; clear i

%%
kegg_recAnoKu = cell(length(recAnoKu), 1);
for i = 1:length(recAnoKu)
    dbentry1 = strcat(dbentry, num2str(recAnoKu(i, 2)));
    kegg_recAnoKu{i} = regexpi(urlread(strcat(base, operation, database,...
        dbentry1)), '(?&lt;=(??@dbentry1)\s+)\w+\W+\w*','match');
end; clear i

%%
kegg_recAwithKu = cell(length(recAwithKu), 1);
for i = 1:length(recAwithKu)
    dbentry1 = strcat(dbentry, num2str(recAwithKu(i, 2)));
    kegg_recAwithKu{i} = regexpi(urlread(strcat(base, operation, database,...
        dbentry1)), '(?&lt;=(??@dbentry1)\s+)\w+\W+\w*','match');
end; clear i

%% QUERY KEGG FOR KU GENE SEQUENCES
base            = 'http://rest.kegg.jp/';
operation       = 'get/';
option          = 'ntseq';

%%
emptyKu = find(cellfun(@isempty,kegg_ku));
kegg_ku(emptyKu) = [];
%     remove all the genes that did not give any hits in KEGG. 
for i = 1:length(kegg_ku)
    kuSeq = urlread(char(strcat(base, operation, kegg_ku{i}, '/', option)));
    headerFind = findstr(') atg', kuSeq);
        if numel(headerFind) == 0
            headerFind = findstr(') gtg', kuSeq);
        elseif numel(headerFind) == 0
            headerFind = findstr(') ctg', kuSeq);
        elseif numel(headerFind) == 0
            headerFind = findstr(') ttg', kuSeq);
        end
    sequenceData = kuSeq(headerFind(1):end);
    fastawrite([basedir 'kuSeq.fasta'], kegg_ku{i},...
        sequenceData);
end; clear i

%%
emptyRecAnoKu = find(cellfun(@isempty,kegg_recAnoKu));
kegg_recAnoKu(emptyRecAnoKu) = [];
%     remove all the genes that did not give any hits in KEGG. 
for i = 1:length(kegg_recAnoKu)
    recAnoKuSeq = urlread(char(strcat(base, operation,...
        kegg_recAnoKu{i}, '/', option)));
    headerFind = findstr('atg', recAnoKuSeq);
        if numel(headerFind) == 0
            headerFind = findstr('gtg', recAnoKuSeq);
        end
    sequenceData = recAnoKuSeq(headerFind(1):end);
    fastawrite([basedir 'recAnoKuSeq.fasta'], kegg_recAnoKu{i},...
        sequenceData);
end; clear i

%%
emptyRecAwithKu = find(cellfun(@isempty,kegg_recAwithKu));
kegg_recAwithKu(emptyRecAwithKu) = [];
%     remove all the genes that did not give any hits in KEGG. 
for i = 1:length(kegg_recAwithKu)
    recAwithKuSeq = urlread(char(strcat(base, operation,...
        kegg_recAwithKu{i}, '/', option)));
    headerFind = findstr('atg', recAwithKuSeq);
        if numel(headerFind) == 0
            headerFind = findstr('gtg', recAwithKuSeq);
        end
    sequenceData = recAwithKuSeq(headerFind(1):end);
    fastawrite([basedir 'recAwithKuSeq.fasta'], kegg_recAwithKu{i},...
        sequenceData);
end; clear i
</code></pre>

<h2 id="cluster-and-calculate-pairwise-distances">Cluster and calculate pairwise distances</h2>
<pre><code class="matlab">% scClusteringPairwiseDistances --- this script will plot the dendrogram
% for the pairwise distance information that Matt obtained for the recA
% sequences.

clc; clear, close all;

basedir = '/Users/Dave/Desktop/recA_ku_orthologSearch/pairwiseDistance/';
%     define the base directory

distanceMatrix = importdata([basedir '151202.csv']);
    distanceMatrix.textdata = distanceMatrix.textdata(2:end, 1:2);
%     import the data

attributes = importdata([basedir 'attributeTable.xlsx']);
    attributes.AttributeFile = attributes.AttributeFile(2:end, :);
%     import the attribute file. Only 'AttributeFile' is of importance.

% %% THIS IS TO CREATE THE CLUSTERGRAM FROM THE DATA
% distanceMatrix.data = flipud(distanceMatrix.data);
% distanceMatrix.textdata = flipud(distanceMatrix.textdata);
% %     data flipped upside down
% 
% data4clustering = squareform(distanceMatrix.data(:, 1));
% %     make the data into a square matrix for clustering
% CGobj = clustergram(data4clustering);
% %     standard clustergram, making an object that is not easily manipulated

%% ATEMPT TO RECREATE THE DENDROGRAM USING LINKAGE AND DENDROGRAM COMMANDS
distanceMatrix.data = flipud(distanceMatrix.data);
distanceMatrix.textdata = flipud(distanceMatrix.textdata);
%     data flipped upside down; this is done to get out the proper format
%     for creating the squareform matrix downstream. it also means that all
%     subsequent data will need to be reorganized. 

labels = flipud(unique(distanceMatrix.textdata));
%     try to get the labels out for the tree and flip for compatibility

data4clustering = squareform(distanceMatrix.data(:, 1));
%     make the data into a square matrix for clustering 

attributes.AttributeFile = flipud(sortrows(attributes.AttributeFile, 1));
%     alphabetically order the attribute file and flip upside down to be
%     compatible with the rest of the data

NHEJ = strfind(attributes.AttributeFile(:, 2), 'NHEJ');
%     denote all NHEJ organisms with 1
NHEJ = ~cellfun(@isempty, NHEJ);
%     replace all empty cells with 0
NHEJ = repmat(NHEJ, 1, 50);

res207 = NaN(length(attributes.AttributeFile), 1);
for i = 1:length(attributes.AttributeFile)
    if strcmp('S', attributes.AttributeFile(i, 3)) == 1;
        res207(i, 1) = 0;
    elseif strcmp('N', attributes.AttributeFile(i, 3)) == 1;
        res207(i, 1) = 1;
    else
        res207(i, 1) = 2;
    end
end; clear i
%     create a vector with different values depending on what is considered
%     residue 207 (S, N, or other).
res207 = repmat(res207, 1, 50);

tree = linkage(data4clustering, 'average', 'euclidean');
%     make the data for the tree. The tree will be made in the exact same
%     fashion as the standard clustergram dendrogram is made. This means
%     'average' linkage and 'euclidian' distance metric.

leafOrder = optimalleaforder(tree, data4clustering);
%     optimize the leaf order.

%%
figure;
    subplot (20, 1, 1:8 )
        dendrogram(tree, 0, 'Labels', [], 'Reorder', leafOrder,...
            'colorthreshold', 75);
            ax = gca;
%             ax.XTickLabelRotation = 90; % make sure the labels are rotated
           set(gca, 'YTick', [], 'XTick', [], 'Box', 'off', 'Color', 'none');
           axis off;
    h = subplot (20, 1, 9:18);
        imagesc(data4clustering(leafOrder, leafOrder));
            set(gca, 'YTick', [], 'XTick', [], 'Color', 'none');
            colormap(h, jet);
    i = subplot (20, 1, 19);
        imagesc(NHEJ(leafOrder))
            set(gca, 'YTick', [], 'XTick', [], 'Color', 'none');
            colormap(i, gray);
    j = subplot (20, 1, 20);
        imagesc(res207(leafOrder))
            set(gca, 'YTick', [], 'XTick', [], 'Color', 'none');
            colormap(j, parula);
clear h i j

%%
N = 0; NKu = 0; S = 0; SKu = 0;
for i = 1:length(attributes.AttributeFile)
    if strcmp(attributes.AttributeFile(i, 3), 'N') &amp;&amp;...
            strcmp(attributes.AttributeFile(i, 2), 'NHEJ')
        NKu = NKu + 1;
    end
    if strcmp(attributes.AttributeFile(i, 3), 'N')
        N = N + 1;
    end
    if strcmp(attributes.AttributeFile(i, 3), 'S') &amp;&amp;...
            strcmp(attributes.AttributeFile(i, 2), 'NHEJ')
        SKu = SKu + 1;
    end
    if strcmp(attributes.AttributeFile(i, 3), 'S')
        S = S + 1;
    end
end; clear i
</code></pre>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="../Loop2/" class="btn btn-neutral" title="Loop2"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../Loop2/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
