<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Loop2 - RecA Loop 2 evolutionary conservation</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Loop2";
    var mkdocs_page_input_path = "Loop2.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> RecA Loop 2 evolutionary conservation</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1 current">
		
    <a class="current" href="./">Loop2</a>
    <ul class="subnav">
            
    <li class="toctree-l2"><a href="#reca-s207-loop2-analysis">RecA S207 Loop2 analysis</a></li>
    
        <ul>
        
            <li><a class="toctree-l3" href="#finding-orthologs">Finding orthologs</a></li>
        
            <li><a class="toctree-l3" href="#figure">Figure</a></li>
        
            <li><a class="toctree-l3" href="#citations">Citations</a></li>
        
        </ul>
    

    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../CODE/">Code</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">RecA Loop 2 evolutionary conservation</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Loop2</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="reca-s207-loop2-analysis">RecA S207 Loop2 analysis</h1>
<h2 id="finding-orthologs">Finding orthologs</h2>
<p>Orthologous genes of M. smegmatis RecA (MSMEG_2723) were obtained through OrtholugeDB, which uses a reciprocal BLAST-based algorithm to ensure validity of the orthologous sequences. Briefly, the top BLAST results from an initial search are BLASTed back against the original reference genome, and the results are only counted as orthologs if the top hit in this second search corresponds to the initial sequence of interest. This approach was applied to both recA and ku. </p>
<p>All reciprocal Ortholuge results lacking a taxonomy ID (yielding it impossible to identify the species of origin bioinformatically) were disregarded. The recA and ku databases were subsequently intersected. Organisms that had both recA and ku coding sequences were considered capable of NHEJ; all others were considered to be capable only of HR. Using taxonomy IDs the KEGG database was then queried for  recA protein-coding nucleotide sequences.</p>
<p>Additionally, we used BLAST and OrtholugeDB to look for organisms that contained Ku, a unique marker of NHEJ in bacteria and archea. This list of organisms was compared with the list of RecA orthologs, and was used to extrapolate whether or not organisms rely solely on HR for DNA repair, or also have NHEJ.</p>
<p>In order to account for both synonymous and non-synonymous mutations, nucleotide sequences for all organisms were translated into protein sequences, then aligned by Muscle, and finally re-translated back to nucleotide sequences using MEGA6. This multiple alignment was used to test 24 nucleotide substitution models, and the model with the lowest Bayesian information criterion (BIC) score was chosen for computing the pairwise evolutionary distances between each sequence—the general time reversible (GTR) model with a gamma distributed substitution rate (Γ) and an allowance for a proportion of invariant sites (I).</p>
<p>The number of base substitutions per site between sequences were used to represent the relative evolutionary distance. Analyses were conducted using the Maximum Composite Likelihood model [1]. The rate variation among sites was modeled with a gamma distribution (shape parameter = 1). The analysis involved 1293 nucleotide sequences. All codon positions were included. All ambiguous positions were removed for each sequence pair. There were a total of 6264 positions in the final dataset. Evolutionary analyses were conducted in MEGA6. </p>
<p>All organisms in the final database were annotated for NHEJ or HR (as mentioned previously), as well as for their amino acid residue corresponding to residue 207 in Msmeg. Using Matlab, a leaf-optimized dendrogram was created using the linkage function from the Statistics Toolbox. For the linkage, ‘average’ (unweighted average distance) was used as the metric, and the Euclidean distance was utilized as the distance metric. </p>
<p>The evolutionary distance matrix was outputted as a heatmap, and two bars were created to indicate the respective DNA repair pathway the organisms presumably possess, as well as the particular residue at the relative position of 207. </p>
<p>See figure for results. </p>
<p>128/787 organisms with RecA-N207 were capable of NHEJ, whereas 146/436 organisms with RecA-S207 were capable of NHEJ (Chi-square test: p &lt;&lt; 0.001).  </p>
<h2 id="figure">Figure</h2>
<p><img alt="HeatmapFigure" src="../img/RecA_orthologs.png" /></p>
<p>Figure: Bioinformatics results suggest that RecA-S207 is correlated with an organism’s capacity to perform NHEJ. At the top of the figure is the dendrogram, based on the evolutionary pairwise distances (second panel: more yellow equals more distant, darker blue equals less distant) calculated from the nucleotide sequences of recA.  The third panel denotes the organism’s capacity to perform NHEJ as judged by the presence of ku in their genome. White denotes that the bacterium is capable of NHEJ (to be changed!!) and black denotes an organism can only perform HR. The lowest panel denotes the respective residue at position 207 (as compared to the Msmeg sequence). </p>
<h2 id="citations">Citations</h2>
<p>OrtholugeDB: Whiteside MD, Winsor GL, Laird MR, Brinkman FS. OrtholugeDB: a bacterial and archaeal orthology resource for improved comparative genomic analysis. Nucleic Acids Research. 2013 Jan;41(Database issue):D366-76.</p>
<p>KEGG: Kanehisa, M., Goto, S. (2000). KEGG: kyoto encyclopedia of genes and genomes. Nucleic Acids Res. 2000 Jan 1;28(1):27-30.</p>
<p>Pairwise distance method: Tamura K., Nei M., and Kumar S. (2004). Prospects for inferring very large phylogenies by using the neighbor-joining method. Proceedings of the National Academy of Sciences (USA) 101:11030-11035.</p>
<p>MEGA6: Tamura K., Stecher G., Peterson D., Filipski A., and Kumar S. (2013). MEGA6: Molecular Evolutionary Genetics Analysis version 6.0. Molecular Biology and Evolution30: 2725-2729.</p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../CODE/" class="btn btn-neutral float-right" title="Code">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href=".." class="btn btn-neutral" title="Home"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href=".." style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../CODE/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
